### PR TITLE
update load_lupus_data function to increase simplicity

### PR DESCRIPTION
**Pull request by pascoesean**
stringing along the anndata object instead of weird function-input-defined slices of it

### CODE REVIEW

**Comment by pascoesean** [comment]
@andrewram4287 do you want to take a look at this? This version should be a lot more simple, I'll just have to add some more processing code elsewhere (which is probably where it should've been anyways...). Only thing with this is that it won't build with the current version of S1 and S2 that are on main; so I can either update them on this PR or just comment them out or something

File: sccp/imports/scRNA.py [hunk 1]
```diff
@@ -135,20 +163,6 @@ def load_lupus_data(third_axis="ind_cov", every_n=4):
     # keep only n observations
     X = X[::every_n, :]
 
-    # get cell types -> only stay true if order of cells doesn't change after this
```
Comment (pascoesean) [inline sccp/imports/scRNA.py:138]: lines 138-151 that I get rid of here are essentially what I add to S1 and S2. that's all that happens

File: sccp/imports/scRNA.py [hunk 2]
```diff
@@ -135,20 +163,6 @@ def load_lupus_data(third_axis="ind_cov", every_n=4):
...
-        .drop_duplicates("ind_cov")
-    )
-
-    lut = {"SLE": "c", "Healthy": "m"}
-    row_colors = status["SLE_status"].map(lut)
-
     assert np.all(np.isfinite(X.X.data))  # this should be true
 
-    return tensorFy(X, third_axis), cell_types, row_colors
+    return tensorFy(X, 'sample_ID'), X.obs
```
Comment (pascoesean) [inline sccp/imports/scRNA.py:168]: I could also just output the entire anndata object `X`, but this (`X.obs`) might take up a bit less room and (at the moment) is the only part I'm actually using

File: sccp/figures/common.py [hunk 3]
```diff
@@ -470,9 +470,9 @@ def plotCellStateViolins(projections, cell_types, cell_state: int, ax):
     cell_state_n.columns.values[0] = "contribution"
 
     sns.violinplot(data = cell_state_n,
-                   x = "cg_cov",
+                   x = "cell_type_broad",
                    y = 'contribution',
-                   hue = 'cg_cov',
+                   hue = 'cell_type_broad',
                    dodge = False,
                    ax = ax)
```
Comment (pascoesean) [inline sccp/figures/common.py:478]: this is just updating names (full dictionary/key in `scRNA.py`)

File: sccp/imports/scRNA.py (hunk 2 above)
Comment (andrewram4287) [reply sccp/imports/scRNA.py:168]: I don't believe there would be a reason to need the X unless you wanted to look at the raw data

File: sccp/figures/figureS2.py [hunk 4]
```diff
@@ -30,21 +30,24 @@ def makeFigure():
     subplotLabel(ax)
 
     # Import of data
-    data, cell_types, _ = load_lupus_data()  # don't need to get patient color mappings
-    rank = 30
-    cellState = 28
-    cmp = 28
+    data, obs = load_lupus_data(every_n=10)  # don't need to get patient color mappings
+    rank = 5
+    cellState = 4
```
Comment (andrewram4287) [inline sccp/figures/figureS2.py:35]: You won't need cell state anymore so you should remove them from your functions

File: sccp/figures/figureS1.py [hunk 5]
```diff
@@ -21,18 +21,27 @@ def makeFigure():
...
+    ) = load_lupus_data(every_n = 10)  # so that it builds quickly
+
 
     weights, factors, _, _ = parafac2_nd(
         lupus_tensor, rank=rank, n_iter_max=20, random_state=1
     )
 
+    status = (
+        obs[["sample_ID", "SLE_status"]]
+        .sort_values(by="sample_ID")
```
Comment (andrewram4287) [inline sccp/figures/figureS1.py:38]: @pascoesean isn't this switching the order that Pf2 orders the patients?

File: sccp/figures/figureS1.py [hunk 6]
```diff
@@ -21,18 +21,27 @@ def makeFigure():
...
+        .drop_duplicates("sample_ID")
+    )
+
+    lut = {"SLE": "c", "Healthy": "m"}
+    row_colors = status["SLE_status"].map(lut)
+
     plotFactors(
         factors, lupus_tensor, ax[0:3], reorder=(0, 2), trim=(2,), row_colors=row_colors
```
Comment (andrewram4287) [inline sccp/figures/figureS1.py:46]: @pascoesean if you are gonna label the patients by status, I believe you have to remove the reorder of "0"

Comment (pascoesean) [reply sccp/figures/figureS1.py:46]: I _think_ this should be ok-- if the reorder tuple includes 0 then it reorders the row_colors along with the patients :

 if i in reorder:
            X, ind = reorder_table(X)
            yt = yt[ind]
            if i == 0 and not (row_colors is None):
                row_colors = row_colors[ind]

Comment (pascoesean) [reply sccp/figures/figureS1.py:46]: ^^ common lines 130-134

File: sccp/figures/figureS1.py (hunk 5 above)
Comment (pascoesean) [reply sccp/figures/figureS1.py:38]: Yes, good catch thank you (I think this is holdover from when I was trying to do something in a separate notebook). It shouldn't make a difference in output though (phew) since the observations are already *mostly* ordered by the scRNA import function itself. I don't see a need to keep this though so I'll take it out

File: sccp/figures/figureS2.py [hunk 7]
```diff
@@ -30,29 +30,31 @@ def makeFigure():
...
+    pf2Points = umap.UMAP(random_state=1, verbose=True).fit(projs)
 
     # PCA dimension reduction
     pc = PCA(n_components=rank)
     pcaPoints = pc.fit_transform(data.unfold())
     pcaPoints = umap.UMAP(random_state=1).fit(pcaPoints)
 
     plotUMAP_ct(cell_types, pf2Points, ax[0])
-    plotCmpUMAP(cellState, cmp, factors, pf2Points, projs, ax[1])
-    plotCellStateViolins(projs, cell_types, cellState, ax[2])
```
Comment (pascoesean) [inline sccp/figures/figureS2.py:56]: this function gets modified and put in S2b. But S2b doesn't exist in this timeline

File: sccp/figures/figureS2.py (hunk 4 above)
Comment (pascoesean) [reply sccp/figures/figureS2.py:35]: @andrewram4287 looking back through this and thinking maybe I should just keep the cell state version in here at least until your PR about proj x B gets merged to main (in which case I could just merge main)-- could also include those changes in this but since that is already a separate PR that might be easier

File: sccp/figures/figureS1.py (hunk 5 above)
Comment (andrewram4287) [reply sccp/figures/figureS1.py]: Yeah, we just want to make sure we are not mixing the order of the patients' statuses

**Comment by pascoesean** [comment]
@aarmey This is an initial PR to get things set up for merging more of my code for the figures I've been working on (we wanted to split things up so it was more manageable to review/merge).

**Review by aarmey — APPROVED** [review]
Looks good to me!
